poc.rb

```diff
@@ -0,0 +1,38 @@
+require 'waterdrop'
+
+producer = WaterDrop::Producer.new
```

Comment:
WDYT about adding ability to also put config into constructor of producer? and use `#setup` too

Reply:
You mean this?
```
producer = WaterDrop::Producer.new do |config|
# trolololo
end
```

We can do that BUT we need to indicate, that this needs to run after fork.

Reply:
yep, something like this. or `producer = WaterDrop::Producer.new({ ... })` if block harder in implementation 🤔

Reply:
already in the 2.0 branch

```ruby
producer = WaterDrop::Producer.new do |config|
  config.kafka = {
    'bootstrap.servers' => 'localhost:9092',
    'request.required.acks' => [-1, 1].sample
  }
end
```